src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ConnectedDevicesSpec.groovy

```diff
@@ -1383,6 +1384,174 @@ srcDevices=#newSrcEnabled, dstDevices=#newDstEnabled"() {
+        initialPropsDst && restoreSwitchProperties(dst.dpId, initialPropsDst)
+    }
+
+    @Tidy
+    def "System stops receiving statistics if config is changed to 'off'"() {
+        given: "A switch with devices feature turned on"
+        def tgService = traffExamProvider.get()
+        assumeTrue(topology.activeTraffGens.size() > 0, "Require at least 1 switch with connected traffgen")
+        def sw = topology.activeTraffGens[0].switchConnected
+        def initialPropsSource = enableMultiTableIfNeeded(true, sw.dpId)
+        switchHelper.updateSwitchProperties(sw, initialPropsSource.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = true
+            it.switchArp = true
+        })
+
+        and: "Flow is created on a target switch with devices feature 'off'"
+        def dst = topology.activeSwitches.find { it.dpId != sw.dpId }
+        def initialPropsDst = enableMultiTableIfNeeded(true, dst.dpId)
+        def flow = flowHelperV2.randomFlow(sw, dst)
+        def outerVlan = 100
+        flow.source.vlanId = outerVlan
+        flow.source.innerVlanId = 200
+
+        flow.source.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flow.destination.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flowHelperV2.addFlow(flow)
+
+        when: "update device properties and send lldp and arp packets on src flow endpoint and match outerVlan only"
+        switchHelper.updateSwitchProperties(sw, initialPropsSource.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = false
+            it.switchArp = false
+        })
+
+        def srcLldpData = LldpData.buildRandom()
+        def dstLldpData = LldpData.buildRandom()
+        def srcArpData = ArpData.buildRandom()
+        def dstArpData = ArpData.buildRandom()
+        [[flow.source, srcLldpData, srcArpData], [flow.destination, dstLldpData, dstArpData]].each {
+            endpoint, lldpData, arpData ->
+                new ConnectedDevice(tgService, topology.getActiveTraffGen(endpoint.switchId), [outerVlan]).withCloseable {
+                    it.sendLldp(lldpData)
+                    it.sendArp(arpData)
+                }
+        }
+
+        then: "Getting connecting devices doesn't show corresponding devices on src endpoint"
+        def response = northboundV2.getConnectedDevices(sw.dpId)
+        sleep(3000)
+        Wrappers.timedLoop(3) {
+            //under usual condition system needs some time for devices to appear, that's why timeLoop is used here
+            verifyAll(northboundV2.getConnectedDevices(sw.dpId)) {
+                (it.ports.lldp.empty)
+            }
+        }
+
+        cleanup: "Delete the flow and restore initial switch properties"
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        initialPropsSource && restoreSwitchProperties(sw.dpId, initialPropsSource)
+        initialPropsDst && restoreSwitchProperties(dst.dpId, initialPropsDst)
```

Comment:
You may still want to cleanup the 'devices' in database here. if the test fails and the system actually shows a connected device, we still want to clean it up so that environment is truly clean. (in cleanup we should always consider to cleanup after failed scenarios as well, not only after successful ones)

Reply:
solved